src/shared/operatorconfig/enforcement/config.go

```diff
@@ -17,6 +18,20 @@ type Config struct {
 	EnableGCPPolicy                      bool
 	EnableAzurePolicy                    bool
 	EnforcedNamespaces                   *goset.Set[string]
+	AllowExternalTraffic                 allowexternaltraffic.Enum
+}
+
+func (c Config) GetActualExternalTrafficPolicy() allowexternaltraffic.Enum {
+	if c.AllowExternalTraffic == allowexternaltraffic.Off {
+		return allowexternaltraffic.Off
+	} else if !c.EnforcementDefaultState && c.AllowExternalTraffic == allowexternaltraffic.Always {
+		// We don't want to create network policies for external traffic when enforcement is disabled.
+		// However, if one uses shadow mode we can still block external traffic to his protected services
+		// therefore we should return allowexternaltraffic.IfBlockedByOtterize
+		return allowexternaltraffic.IfBlockedByOtterize
+	} else {
+		return c.AllowExternalTraffic
+	}
```

Comment:
consider using switch, plus this would be less confusing IMO:
```
switch c.AllowExternalTraffic {
case allowexternaltraffic.Off: return allowexternaltraffic.Off
case allowexternaltraffic.Always:
    if c.EnforcementDefaultState { return allowexternaltraffic.Always }
    else {return allowexternaltraffic.IfBlockedByOtterize} // and the long documentation goes here
default: // allowexternaltraffic.IfBlockedByOtterize
   return allowexternaltraffic.IfBlockedByOtterize
}
}
```